### DISCUSSION TITLE
Improving consistency of include guards

### DISCUSSION THREAD

**Discussion by cflaviu** [managarm/managarm#308] [Ideas and RFCs]
The include guards in project's headers are either classic #ifndef-#define-#endif guard or #pragma once. Only one type of include guard should be used.

**Comment by ArsenArsen** [comment]
The latter is preferred and should be used in new code, as far as I understand.

+1 on making this consistent (clang-tidy rules perhaps?)

**Reply by cflaviu** [reply]
I prefer #pragma once too despite the fact I don't know when this include guard may fail.

**Reply by Dennisbonke** [reply]
#pragma once is in fact preferred, we settled this in the coding standards document found [here](https://docs.managarm.org/handbook/contributing/coding-style.html).

**Reply by cflaviu** [reply]
PR created
